ui/apps/platform/src/Containers/NetworkGraph/components/NetworkBreadcrumbs.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, ChangeEvent } from 'react';
+import {
+    Breadcrumb,
+    BreadcrumbItem,
+    Select,
+    SelectOption,
+    SelectVariant,
+} from '@patternfly/react-core';
+
+import { Cluster } from 'types/cluster.proto';
+import useURLSearch from 'hooks/useURLSearch';
+import useFetchClusterNamespaces from 'hooks/useFetchClusterNamespaces';
+import { ClusterIcon, NamespaceIcon } from '../common/NetworkGraphIcons';
+import getScopeHierarchy from '../utils/getScopeHierarchy';
+
+function filterElementsWithValueProp(
+    filterValue: string,
+    elements: React.ReactElement[] | undefined
+): React.ReactElement[] | undefined {
+    if (filterValue === '' || elements === undefined) {
+        return elements;
+    }
+
+    return elements.filter((reactElement) =>
+        reactElement.props.value?.toLowerCase().includes(filterValue.toLowerCase())
+    );
+}
+
+type NetworkBreadcrumbsProps = {
+    clusters: Cluster[];
+};
+
+function NetworkBreadcrumbs({ clusters = [] }: NetworkBreadcrumbsProps) {
+    const [isClusterOpen, setIsClusterOpen] = React.useState(false);
+    const [isNamespaceOpen, setIsNamespaceOpen] = React.useState(false);
+
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    const {
+        cluster: selectedClusterName,
+        namespaces: selectedNamespaces,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        deployments: deploymentsFromUrl,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        remainingQuery,
+    } = getScopeHierarchy(searchFilter);
+    const selectedClusterId =
+        clusters.find((cluster) => cluster.name === selectedClusterName)?.id || '';
+    const { loading, error, namespaces } = useFetchClusterNamespaces(selectedClusterId);
+
+    const onClusterToggle = (newClusterOpenValue) => setIsClusterOpen(newClusterOpenValue);
```

Comment:
We do have one that I think fits this use case: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/hooks/patternfly/useSelectToggle.ts

It is unfortunately named `useSelectToggle`, when in reality is should probably be `useToggle` since it can be used as a generic toggle for things other than `<Select>` components.